src/index.js

```diff
@@ -82,12 +82,12 @@ const getEntities = (entities, isImmutable) => (schema, entityOrId) => {
   return getEntity(entityOrId, schemaKey, entities, isImmutable);
 };
 
-export const denormalize = (input, schema, entities) => {
+export const denormalize = (input, schema, entities, cache = {}) => {
   if (!input) {
     return input;
   }
 
   const isImmutable = ImmutableUtils.isImmutable(entities);
   const getDenormalizedEntity = getEntities(entities, isImmutable);
```

Comment:
The `cache` object should be passed to `getEntities`. The return function for `getEntities` should do the cache logic instead of `schema.Entity`. That'd reduce the amount of passing `cache` around to all of these other methods and centralize the logic in case of future `Entity` types.